### ISSUE TITLE
the value of the model.fit_transform() is nan

### ISSUE THREAD

**Issue by nieOneq** [jameschapman19/cca_zoo#173]
Dear James,
 Thank you very much for your work, but I have some confusion on some parameter settings
 
1.When the number of branches is 2000, the value of the model.fit_transform()  is normal, and when the number of branches is 200, the value is nan.

2. Changing the value of the LATENT_DIMS will have the same result in the same case.

`#datasetName:3sources,MSRC_v1,BBCSport,BBC,ORL,handwritten,Caltech7
dataName='handwritten'
dataset=loadDatasets(dataName)
data,label=dataset.get_data,dataset.get_label#nadarray lists
label_pipeline = lambda k: k - 1
label=label_pipeline(label)

for i in range(len(data)):
    data[i][data[i]==0]+=1e-8

#------------------------change data type for DCCA-----------------------
data_dict=NumpyDataset([data[0],data[1],data[2],data[3],data[4]],labels=label)
check_dataset(data_dict)
data_loader=get_dataloaders(data_dict,batch_size=2000)

LATENT_DIMS = 50
EPOCHS = 10

encoders = [architectures.Encoder(latent_dims=LATENT_DIMS, feature_size=int(data[i].shape[1])) for i in range(5)]

# %%
# Deep MCCA
# ----------
dcca = DCCA(
    latent_dims=LATENT_DIMS, encoders=encoders, objective=objectives.MCCA,
)
trainer = pl.Trainer(max_epochs=EPOCHS)
trainer.fit(dcca, data_loader)
t=dcca.transform(data_loader)
`

**Comment by jameschapman19** [comment]
Hi can you define branches for me?

**Comment by nieOneq** [comment]
Thank you very much for your quick reply, I will put a screenshot of the sample structure below

![image](https://user-images.githubusercontent.com/101241575/231402472-a122d72a-fd05-401f-a650-a724045d0b17.png)

**Comment by nieOneq** [comment]
Hello I don't know if there are any conditional restrictions in terms of the two parameters of branch and output dimension, I tried some methods to avoid nan, but none of them worked. After the sixth view is added, no matter how the two parameters of branch and output dimension are set, the last is nan.
![1681289087352](https://user-images.githubusercontent.com/101241575/231404230-ecbfa787-0744-40f0-8aa5-a3e21ec21921.png)

**Comment by nieOneq** [comment]
> Hi can you define branches for me?

Thank you very much for your work, I followed the documentation instructions of cca-zoo to learn, I wonder if it is that I did not notice some parameter limitation issues.

**Comment by nieOneq** [comment]
> Hi can you define branches for me?
I'm very sorry that the branch is that I didn't express it clearly, it means that the parameters in the get_dataloaders are batch_size, is there a numerical limit for batch_size and LATENT_DIMS?

**Comment by nieOneq** [comment]
> Hi can you define branches for me?

I'm very sorry that the branch is that I didn't express it clearly, it means that the parameters in the get_dataloaders are batch_size, is there a numerical limit for batch_size and LATENT_DIMS?
